rootfs-builder/rootfs.sh

```diff
@@ -0,0 +1,147 @@
+#!/bin/bash
+#
+# Copyright (c) 2017 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+set -e
+
+script_name="${0##*/}"
+script_dir="$(dirname $(realpath -s $0))"
+ROOTFS_DIR=${ROOTFS_DIR:-${PWD}/rootfs}
+AGENT_VERSION=${AGENT_VERSION:-master}
+GO_AGENT_PKG=${GO_AGENT_PKG:-github.com/kata-containers/agent}
+AGENT_BIN=${AGENT_BIN:-kata-agent}
+
+if [ -n "$DEBUG" ] ; then
+	set -x
+fi
+
+usage(){
+	cat <<EOT
+USAGE: Build a Guest OS rootfs for Kata Containers image
+${script_name} [options] <distro_name>
+
+<distro_name> : Linux distribution to use as base OS.
+
+Supported Linux distributions:
+
+$(get_distros)
+
+Options:
+-a  : agent version DEFAULT: ${AGENT_VERSION} ENV: AGENT_VERSION 
+-h  : Show this help message
+-r  : rootfs directory DEFAULT: ${ROOTFS_DIR} ENV: ROOTFS_DIR
+
+ENV VARIABLES:
+GO_AGENT_PKG: Change the golang package url to get the agent source code
+            DEFAULT: ${GO_AGENT_PKG}
+AGENT_BIN   : Name of the agent binary (needed to check if agent is installed)
+USE_DOCKER: If set will build rootfs in a Docker Container (requries docker)
+            DEFAULT: not set
+EOT
+	exit 1
```

Comment:
Pedantic nit: Calling the `usage` function isn't always an error (`-h` for example).